source/patterns/@aws-solutions-constructs/aws-cloudfront-oai-s3/lib/index.ts

```diff
@@ -0,0 +1,223 @@
+  /**
+   * Existing instance of S3 Content Bucket object, providing both this and `bucketProps` will cause an error.
+   *
+   * @default - None
+   */
+  readonly existingBucketObj?: s3.IBucket,
+  /**
+   * Optional user provided props to override the default props for the S3 Content Bucket.
+   *
+   * @default - Default props are used
+   */
+  readonly bucketProps?: s3.BucketProps,
+
+  // =====================
+  // S3 Content Bucket Access Logs Bucket
+  // =====================
+  /**
+   * Optional - Whether to maintain access logs for the S3 Content bucket
+   *
+   * @default - true
+   */
+  readonly logS3AccessLogs?: boolean,
+  /**
+   * Optional user provided props to override the default props for the S3 Content Bucket Access Log Bucket.
+   *
+   * @default - Default props are used
+   */
+  readonly loggingBucketProps?: s3.BucketProps,
+
+  // =====================
+  // CloudFront Log Bucket
+  // =====================
+  /**
+   * Optional user provided props to override the default props for the CloudFront Log Bucket.
+   *
+   * @default - Default props are used
+   */
+  readonly cloudFrontLoggingBucketProps?: s3.BucketProps,
+
+  // =====================
+  // CloudFront Logs Bucket Access Log Bucket
+  // =====================
+  /**
+   * Optional - Whether to maintain access logs for the CloudFront Logging bucket. Specifying false for this
+   * while providing info about the log bucket will cause an error.
+   *
+   * @default - true
+   */
+  readonly logCloudFrontAccessLog?: boolean,
+  /**
+   * Optional user provided props to override the default props for the CloudFront Log Bucket Access Log bucket.
+   * Providing both this and `existingcloudFrontLoggingBucketAccessLogBucket` will cause an error
+   *
+   * @default - Default props are used
+   */
+  readonly cloudFrontLoggingBucketAccessLogBucketProps?: s3.BucketProps,
+}
+
+export class CloudFrontToOaiToS3 extends Construct {
+  public readonly cloudFrontWebDistribution: cloudfront.Distribution;
+  public readonly cloudFrontFunction?: cloudfront.Function;
+  public readonly cloudFrontLoggingBucket?: s3.Bucket;
+  public readonly cloudFrontLoggingBucketAccessLogBucket?: s3.Bucket;
+  public readonly s3BucketInterface: s3.IBucket;
+  public readonly s3Bucket?: s3.Bucket;
+  public readonly s3LoggingBucket?: s3.Bucket;
+  public readonly originAccessControl?: cloudfront.CfnOriginAccessControl;
+
+  /**
+   * @summary Constructs a new instance of the CloudFrontToOaiToS3 class.
+   * @param {Construct} scope - represents the scope for all the resources.
+   * @param {string} id - this is a a scope-unique id.
+   * @param {CloudFrontToOaiToS3Props} props - user provided props for the construct
+   * @since 0.8.0
+   * @access public
+   */
+  constructor(scope: Construct, id: string, props: CloudFrontToOaiToS3Props) {
+    super(scope, id);
+
+    defaults.printWarning(`This construct deploys a Cloudfront/S3 pattern connected with an Origin Access Identity,
```

Comment:
I like this warning. It is probably enough. We would never want to enforce it more strictly, would we? Like by looking at the partition value and throwing an error?

Reply:
I thought about that, but was concerned about future partitions. Plus, while I can't think of one, someone may have a legit reason to use an OAI elsewhere.